apis/mutations/v1beta1/externaldata.go

```diff
@@ -0,0 +1,38 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import "github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
+
+// ExternalData describes the external data source to use for the mutation.
+type ExternalData struct {
+	// Provider is the name of the external data provider.
+	// +kubebuilder:validation:Required
+	Provider string `json:"provider,omitempty"`
+
+	// DataSource specifies where to extract the data that will be sent
+	// to the external data provider as parameters.
+	// +kubebuilder:default="ValueAtLocation"
+	DataSource types.ExternalDataSource `json:"dataSource,omitempty"`
+
+	// FailurePolicy specifies the policy to apply when the external data
+	// provider returns an error.
+	// +kubebuilder:default="Fail"
```

Comment:
I forgot where we said the default value should be "Fail". Is that what we want? @sozercan @maxsmythe?

Reply:
Yeah, would have to be "fail". The other option is "default", and I don't think we could pick a universally good value for a default fallback.

Reply:
Another option is `Ignore` too but then user wouldn't know if it failed or not. I think `Fail` works as a default value.

Reply:
Ah yeah, forgot about ignore